msa/js-executor/api/jsInvokeMessageProcessor.js

```diff
@@ -186,11 +193,11 @@ JsInvokeMessageProcessor.prototype.sendResponse = function (requestId, responseT
     );
 }
 
-JsInvokeMessageProcessor.prototype.getOrCompileScript = function(scriptId, scriptBody) {
+JsInvokeMessageProcessor.prototype.getOrCompileScript = function (scriptId, scriptBody) {
     var self = this;
-    return new Promise(function(resolve, reject) {
-        if (self.scriptMap[scriptId]) {
-            resolve(self.scriptMap[scriptId]);
+    return new Promise(function (resolve, reject) {
+        if (self.scriptMap.has(scriptId)) {
```

Comment:
You are searching script in map two times. One for "has" and one for "get" method call. This  can be optimized to one "get" and analyze the result.

Reply:
> You are searching script in map two times. One for "has" and one for "get" method call. This can be optimized to one "get" and analyze the result.

fixed